geoips_clavrx/plugins/modules/readers/clavrx_hdf4.py

```diff
@@ -175,3 +176,21 @@ def call(fnames, metadata_only=False, chans=None, area_def=None, self_register=F
     # istat, outputs= read_cloudprops(fname, minlon, maxlon, minlat, maxlat)
     xarrays = read_cloudprops(fname, chans=chans, metadata_only=metadata_only)
     return {"DATA": xarrays, "METADATA": xarrays[[]]}
+
+
+def get_test_files(test_data_dir):
+    """Return test xarray and files for unit testing."""
+    test_file = join(
+        test_data_dir,
+        "test_data_clavrx",
+        "data",
+        "himawari9_2023101_0300",
+        "clavrx_H09_20230411_0300_B01_FLDK_DK_R10_S0110.DAT.level2.hdf",
+    )
+    tmp_xr = call([test_file])
+    return tmp_xr
```

Comment:
How can we tell from the pytest output if this test was actually run?

Reply:
@mindyls @AJThorpe When using `pytest.mark.parametrize()`, pytest will use the arguments to generate IDs if the arguments are strings. When the arguments aren't strings, though, it will just generate test IDs that are `f"{argument_name}{argument_index}"`. So, this:

```python
@pytest.mark.parametrize("plugin", yield_plugins(), ids=generate_test_ids)
def test_is_plugin_valid(plugin):
    ...
```

results in this:

```
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin102] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin103] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin104] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin105] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin106] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin107] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin108] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin109] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin110] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin111] PASSED
tests/unit_tests/plugins/yaml/test_all_yaml_plugins.py::test_is_plugin_valid[plugin112] PASSED
```

To fix this we can create a callback function to use in the call to `pytest.mark.parametrize()`. The function takes in the arguments that are passed to a single call to the test function. It must manipulate the info from the arguments to return a string. Something like this:

```python
def generate_test_ids(val):
    print(f"In GenerateTestIDs: {val}")
    return f"{val['interface']}.{val['name']}"


@pytest.mark.parametrize("plugin", yield_plugins(), ids=generate_test_ids)
def test_is_plugin_valid(plugin):
    ...
```

results in this:

```
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[product_defaults.bases.colormapper-plugin9] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[product_defaults.bases.colormapper-plugin10] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[product_defaults.bases.colormapper-plugin11] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[product_defaults.bases.colormapper-plugin12] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[product_defaults.bases.colormapper-plugin13] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[product_defaults.bases.algorithm-plugin14] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[product_defaults.bases.algorithm-plugin15] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[product_defaults.bases.coverage_checker-plugin16] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[product_defaults.bases.coverage_checker-plugin17] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[feature_annotators.cartopy-plugin18] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[feature_annotators.cartopy-plugin19] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[feature_annotators.cartopy-plugin20] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[products.single-plugin21] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[products.single-plugin22] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[products.single-plugin23] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[products.bases.product-plugin24] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[products.bases.product-plugin25] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[bases.valid_identifier-abi] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[bases.valid_identifier-abi_infrared] PASSED
tests/unit_tests/schema/test_yaml_schema.py::test_is_plugin_valid[bases.valid_identifier-abi_infrared_123] PASSED
```

Reply:
I think this can be handled in a new issue. @AJThorpe Can you create a new issue to add this kind of callback to all of our parametrized tests, then resolve this thread?

Reply:
@jsolbrig I think we can do the same with just increasing the verbosity, here is what the output looks like:

```
$ pytest -v --cov=geoips/geoips  geoips/geoips/tests/unit_tests_long/
======================================================== test session starts =========================================================platform linux -- Python 3.10.12, pytest-7.4.2, pluggy-1.3.0 -- /home/ajthorp/mambaforge/envs/geoips/bin/python3.10
cachedir: .pytest_cache
rootdir: /home/ajthorp
plugins: anyio-4.0.0, cov-4.1.0
collected 30 items

geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[abi_l2_netcdf] XFAIL                                                                          [  3%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[abi_netcdf] XFAIL                                                                             [  6%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[ahi_hsd] XFAIL                                                                                [ 10%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[amsr2_netcdf] PASSED                                                                          [ 13%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[amsr2_remss_winds_netcdf] PASSED                                                              [ 16%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[amsub_hdf] XFAIL                                                                              [ 20%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[amsub_mirs] XFAIL                                                                             [ 23%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[ascat_uhr_netcdf] XFAIL                                                                       [ 26%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[atms_hdf5] XFAIL                                                                              [ 30%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[ewsg_netcdf] XFAIL                                                                            [ 33%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[geoips_netcdf] XFAIL                                                                          [ 36%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[gmi_hdf5] PASSED                                                                              [ 40%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[imerg_hdf5] XFAIL                                                                             [ 43%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[mimic_netcdf] XFAIL                                                                           [ 46%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[modis_hdf4] XFAIL                                                                             [ 50%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[saphir_hdf5] XFAIL                                                                            [ 53%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[sar_winds_netcdf] PASSED                                                                      [ 56%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[scat_knmi_winds_netcdf] PASSED                                                                [ 60%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[scat_noaa_winds_netcdf] PASSED                                                                [ 63%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[seviri_hrit] XFAIL                                                                            [ 66%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[sfc_winds_text] XFAIL                                                                         [ 70%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[smap_remss_winds_netcdf] PASSED                                                               [ 73%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[smos_winds_netcdf] XFAIL                                                                      [ 76%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[ssmi_binary] XFAIL                                                                            [ 80%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[ssmis_binary] XFAIL                                                                           [ 83%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[viirs_netcdf] XFAIL                                                                           [ 86%]
geoips/geoips/tests/unit_tests_long/plugins/modules/readers/test_readers.py::TestReaders::test_reader_plugins[wfabba_ascii] .......
```

Reply:
That looks like a much simpler way to handle it :)

Maybe need to do the more complicated thing for the YAML-based plugin tests, but module based may work just fine by passing -v ?

Reply:
Yes, `-v/--verbose` solves this issue here. I guess I got a little caught up in solving the problem but forgot what the problem was to begin with. Scope creep!

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/issues/394
https://github.com/NRLMMD-GEOIPS/geoips/issues/395